### PR TITLE
KG 18: improve parameter name

### PR DESCRIPTION
**Pull request by kevgo**
__BREAKING CHANGE:__ changes the name of the environment variable that configures service routes from SERVICE_MESSAGES to SERVICE_ROUTES

_Messages_ are actual messages sent between services. _Routes_ is the information which _service type_ can send which _message type_.

@trushton @hugobho

### CODE REVIEW

File: exocom-server/src/client-registry/client-registry.ls [hunk 1]
```diff
@@ -38,19 +38,20 @@ class ClientRegistry
...
-  # Adds service routing configurations to the given setup
-  register-client: (service) ->
-    @clients[service.name] =
-      client-name: service.name
-      service-type: service.name
-      internal-namespace: @routing[service.name].internal-namespace
+  # registers the given service instance that just came online
+  register-client: (client) ->
+    @clients[client.name] =
+      client-name: client.name
```
Comment (trushton) [inline exocom-server/src/client-registry/client-registry.ls]: Do we need a step here to make sure that the client name is available, and to alter it if it isn't? In #26 you mention that the client names could be essentially the `service-type` followed by an incremental number, e.g. `web1`, `web2`.

File: exocom-server/CONTRIBUTING.md [hunk 2]
```diff
@@ -11,6 +11,18 @@ The default port is 3100.
 More details around configuring the port [here](features/configuring-the-ports.feature).
 
 
+## Terminology
+
+- there are a number of __service types__, for example the "web" service, the "todo" service, etc
+- each service type has a number of instances. We call them __clients__ of Exocom to avoid the word "service" again.
+- each client has a unique name to identify it, e.g. "web #1"
+- the service configuration file defines which messages a service is allowed to send and receive.
+  This information is called __routing__
```
Comment (trushton) [inline exocom-server/CONTRIBUTING.md]: In consistency with this, should the `messages` section of a service's `service.yml` be renamed to `routing`?

Comment (hugobho) [inline exocom-server/CONTRIBUTING.md]: I think a service stills sends individual `messages`, but `routing` is a list of what messages a service can send or receive.

Comment (kevgo) [inline exocom-server/CONTRIBUTING.md]: +1 on what @hugobho said. `service.yml` defines the messages that a service sends and receives. Routing is the table compiled out of that information that says which message should be forwarded to which roles.

File: exocom-server/src/client-registry/client-registry.ls (hunk 1 above)
Comment (kevgo) [inline exocom-server/src/client-registry/client-registry.ls]: Yes, absolutely! Let's do that once we build support for multiple instances of roles. That feature needs proper feature specs, tests, etc, so I don't want to do that here in this PR.

**Pull request closed without merging**